README.md

```diff
@@ -26,3 +31,79 @@ For this [wasm-pack](https://rustwasm.github.io/wasm-pack/) is used.
+* `config.plural_count`: `2` for base language.
+* `base`: Base string to validate
+
+**Output:**
+* `errors`: List of errors. If this is not empty, the string should not be offered to translators.
+* `normalized`: The normalized text is the text, which should be displayed to translators.
+    * In the normalized text, string commands like `RAW_STRING`, `STRING5`, ... are replaced with `STRING`.
+    * Translators can copy the normalized text as template for their translation.
+
+**Example:**
+```console
+>>> cargo run "{BLACK}Age: {LTBLUE}{STRING2}{BLACK}   Running Cost: {LTBLUE}{CURRENCY}/year"
+ERROR at byte 61: Unknown string command '{CURRENCY}'.
+
+>>> cargo run "{BLACK}Age: {LTBLUE}{STRING2}{BLACK}   Running Cost: {LTBLUE}{CURRENCY_LONG}/year"
+NORMALIZED:{BLACK}Age: {LTBLUE}{0:STRING}{BLACK}   Running Cost: {LTBLUE}{1:CURRENCY_LONG}/year
+```
+
+### Step 2: Translators enter stuff
+
+* Translators must provide a text for the default case.
+* Other cases are optional.
+* Game-scripts do not support cases. There is a method in `LanguageConfig` to test for this, but it is not exported yet.
+
+### Step 3: Validate and normalize the translation string
+
+**API method:**
+```rust
+fn validate_translation(config: LanguageConfig, base: String, case: String, translation: String) -> ValidationResult
+```
+
+**Input:**
+* `config.dialect`: One of `openttd`, `newgrf`, `game-script`.
+* `config.cases`: `case` from `nile-config`.
+* `config.genders`: `gender` from `nile-config`.
+* `config.plural_count`: Number of plural forms from `nile-config`.
+* `base`: Base string the translation is for.
+* `case`: Case for the translation. `""` and `"default"` are both valid for the default case.
+* `translation`: The text entered by the translator.
+
+**Output:**
+* `errors`: List of errors.
+    * `critical`: Severity of the error.
+        * `true`: The translation is broken, and must not be committed to OpenTTD.
+        * `false`: The translation is okay to commit, but translators should fix it anyway. This is used for new validations, which Eints did not do. So there are potentially lots of existing translations in violation.
+    * `position`: Byte position in input string. `None`, if general message without location.
+    * `message`: Error message.
+    * `suggestion`: Some extended message with hints.
+* `normalized`: The normalized text is the text, which should be committed. In the normalized text, trailing whitespace and other junk has been sanitized.
+
+**Example:**
+```console
+>>> cargo run "{BLACK}Age: {LTBLUE}{STRING2}{BLACK}   Running Cost: {LTBLUE}{CURRENCY_LONG}/year" "{BLUE}Alter: {LTBLUE}{STRING}{BLACK} Betriebskosten: {LTBLUE}{0:CURRENCY_LONG}/Jahr"
+ERROR at byte 61: Duplicate parameter '{0:CURRENCY_LONG}'.
+ERROR at byte 61: Expected '{0:STRING2}', found '{CURRENCY_LONG}'.
+ERROR: String command '{1:CURRENCY_LONG}' is missing.
+WARNING: String command '{BLUE}' is unexpected.
+HINT: Remove this command.
```

Comment:
It is very unclear what `hints` refer to, but I guess that is just the outputting. I would suggest:

`WARNING: String command '{BLUE}' is unexpected. Hint: Remove this command.`

As now it reads like `ERROR` / `WARNING` / `HINT` are the severity, which makes little sense :)